plugins/woocommerce/includes/wc-core-functions.php

```diff
@@ -1306,6 +1306,50 @@ function wc_get_base_location() {
 	return wc_format_country_state_string( $default );
 }
 
+/**
+ * Uses geolocation to get the customer country and state only if they are valid values.
+ *
+ * @since 9.5.0
+ * @param array $fallback Fallback location.
+ * @return array
+ */
+function wc_get_customer_geolocation( $fallback = array(
+	'country' => '',
+	'state'   => '',
+) ) {
+	$ua = wc_get_user_agent();
+
+	// Exclude common bots from geolocation by user agent.
+	if ( stristr( $ua, 'bot' ) || stristr( $ua, 'spider' ) || stristr( $ua, 'crawl' ) ) {
```

Comment:
~~By "common bots" is the case-sensitive check deliberate? I see some examples [here](https://github.com/monperrus/crawler-user-agents/blob/master/crawler-user-agents.json) for "Bot". 
Can we use stristr() or stripos()?~~ Misread the code as `strstr()` 🙄

Nevertheless I think `stripos() !== false` might be faster, there's a note on it on the [PHP Manuel](https://www.php.net/manual/en/function.strstr.php) (`str_contains()` is PHP 8 only I believe)

Reply:
PHP Manuel 🇲🇽 😂

Reply:
Note `stristr` is case-insenstiive, `stripos` is not so we'd have to call 2 functions. I will leave it; this is not new code and I don't want to break it :D